packages/react-native-codegen/src/CodegenSchema.js

```diff
@@ -299,6 +299,7 @@ export type NativeModuleTypeAliasTypeAnnotation = $ReadOnly<{
 
 export type NativeModulePromiseTypeAnnotation = $ReadOnly<{
   type: 'PromiseTypeAnnotation',
+  elementType?: Nullable<NativeModuleBaseTypeAnnotation>,
```

Comment:
For my understanding -- how it works by adding this elementType to support Promise generics?

Reply:
`CodegenSchema.js` is just a file that define types.

The `NativeModulePromiseTypeAnnotation` is the type we use to describe a promise. The previous implementation was implying that the promise was containing any kind of value. We could have translated with just `Promise` which is invalid even in Javascript. 

By adding the `elementType`, we can now have a `Promise` type which can Host a specific Type, like `Promise<string>`.

The question is: why are we allowing the `elementType` to not be present? Is it for backward compatibility?
As far as I know, all the interfaces we define must return `Promise<T>`, we cannot have an API that returns just a `Promise`, or am I missing something?

Reply:
I thought about that as well and thought it was for backward compatibility. But is that going to be an issue if we remove the optional null types?

Reply:
> But is that going to be an issue if we remove the optional null types?

It will prevent backends to generate a more accurate type in native languages.